tests/suite/graceful_recovery_test.go

```diff
@@ -100,8 +102,122 @@ var _ = Describe("Graceful Recovery test", Ordered, Label("functional", "gracefu
 		Skip("Test currently fails due to this issue: https://github.com/nginxinc/nginx-gateway-fabric/issues/1108")
 		runRecoveryTest(teaURL, coffeeURL, ngfPodName, nginxContainerName, files, &ns)
 	})
+
+	It("recovers when drained node is restarted", func() {
+		runRestartNodeTest(teaURL, coffeeURL, files, &ns, true)
+	})
+
+	It("recovers when node is restarted abruptly", func() {
+		// FIXME(bjee19) remove Skip() when https://github.com/nginxinc/nginx-gateway-fabric/issues/1108 is completed.
+		Skip("Test currently fails due to this issue: https://github.com/nginxinc/nginx-gateway-fabric/issues/1108")
+		runRestartNodeTest(teaURL, coffeeURL, files, &ns, false)
+	})
 })
 
+func runRestartNodeTest(teaURL, coffeeURL string, files []string, ns *core.Namespace, drain bool) {
+	nodeNames, err := getNodeNames()
+	Expect(err).ToNot(HaveOccurred())
+	Expect(nodeNames).To(HaveLen(1))
```

Comment:
I think that since we are running this test on Kind instead of GKE such that we can run it in the pipeline with the functional tests, we are guaranteed to have one node for this test. This test doesn't need to be run on Kind, though that is what I was going off of.

Reply:
I think based on the way the test is written, it does need to be run on kind. For example, the test assumes the k8s control plane runs as a docker container. This is only the case for kind and other kubernetes on docker solutions.

Reply:
Oh I see, ok! Should I leave a comment anywhere about that or is that not needed?

Reply:
I definitely think we should add a comment, maybe even a label. Do you know if there are any other tests that require a certain k8s distro?

Reply:
The NFR tests ensure we run on GKE otherwise they are skipped. We could just do the opposite for this test.

Reply:
I like that idea

Reply:
I think I did this correctly by adding a Skip in the BeforeSuite. Since we currently only test on GKE or on Kind I just checked if we were on GKE then if we were, and the label filter was for "graceful-recovery" we would skip. 

Also, https://onsi.github.io/ginkgo/#skipping-specs confirms that a Skip in the BeforeSuite would skip the entire suite.

Reply:
LGTM